### PR TITLE
fix paste freeze (#1366)

### LINKED ISSUE #1366: Pasting certain text from clipboard into vim makes zellij freeze
**In Case of Graphical or Performance Issues**

1. Delete the contents of `/tmp/zellij-1000/zellij-log`, ie with `cd /tmp/zellij-1000/` and `rm -fr zellij-log/`
2. Run `zellij --debug`
3. Recreate your issue.
3. Quit Zellij immediately with ctrl-q (your bug should ideally still be visible on screen)

Please attach the files that were created in `/tmp/zellij-1000/zellij-log/` to the extent you are comfortable with.

[zellij.log](https://github.com/zellij-org/zellij/files/8603420/zellij.log)
[zellij-11.log](https://github.com/zellij-org/zellij/files/8603423/zellij-11.log)

**Basic information**

`zellij --version`: 0.29.0, but happened on 0.28.1 as well
`tput lines`: 30
`tput cols`: 237
`uname -av` or `ver`(Windows): `Linux XXXXX 5.17.4-200.fc35.x86_64 #1 SMP PREEMPT Wed Apr 20 15:37:53 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

List of programs you interact with as, `PROGRAM --version`: output cropped meaningful, for example:
- `vim --version`: `Vi IMproved 8.2 (2019 Dec 12, compiled Apr 22 2022 00:00:00)`

**Further information**
Reproduction steps, noticeable behavior, related issues, etc

- Open any empty file in vim
- Enter insert mode
- Paste the content of the GNU GPLv3
- zellij freezes and never comes back

### PR DESCRIPTION
**Pull request by tlinford**
Resolve #1366 by offloading writes to a separate thread.

### CODE REVIEW

**Comment by raphCode** [comment]
Can confirm that the deadlock is not happening with this PR, good work!

**Review by imsnif — CHANGES_REQUESTED** [review]
Hey @tlinford - thanks for the great troubleshooting on this!

I left some comments, mostly stylistic but I'm also unsure why we increased the channel bound by so much. I haven't tested this yet because I wanted to sort through that issue first. Once we do, I'll give this a thorough hammering and we'll make sure it's all good.

File: zellij-server/src/lib.rs [hunk 1]
```diff
@@ -575,14 +579,19 @@ fn init_session(
     let to_screen = SenderWithContext::new(to_screen);
 
     let (to_screen_bounded, bounded_screen_receiver): ChannelWithContext<ScreenInstruction> =
-        channels::bounded(50);
+        channels::bounded(5000);
```
[the remark this answers is not in the document]
Comment (tlinford) [reply zellij-server/src/lib.rs:582]: another oops, a leftover from my previous debuggin attempts

**Comment by tlinford** [comment]
@imsnif cleaned it up, should be good for testing.

**Comment by raphCode** [comment]
Seems to work for me!

**Pull request merged**